apollo/build.gradle.kts

```diff
@@ -31,14 +31,14 @@ apollo {
         generateOptionalOperationVariables.set(false)
 
         mapScalarToKotlinString("URL")
-        mapScalar("LocalDate", "java.time.LocalDate")
-        mapScalarToUpload("Upload")
-        mapScalar("PaymentMethodsResponse", "com.adyen.checkout.components.model.PaymentMethodsApiResponse")
         mapScalarToKotlinString("CheckoutPaymentsAction")
         mapScalarToKotlinString("CheckoutPaymentAction")
-        mapScalar("JSONString", "org.json.JSONObject")
-        mapScalar("Instant", "java.time.Instant")
+        mapScalarToUpload("Upload")
+        mapScalar("LocalDate", "java.time.LocalDate", "com.apollographql.apollo3.adapter.JavaLocalDateTimeAdapter")
```

Comment:
Is this one as promiscuous as the one we implemented? If not, we may run into problems here.

Reply:
Right, I've been meaning to ask you about this. I wasn't quite sure what `promiscuous` meant in this context. I could re-create the same functionality in a new adapter though, was the idea that we were not interested at all in all the content of the string after the YYYY-MM-DD ?
So reading a string like `2011-01-02T05:19` as just `2011-01-02`.
If we're relying on that behavior somewhere somehow it should be easy to just bring that custom adapter back in.

Reply:
<img width="646" alt="image" src="https://user-images.githubusercontent.com/39644124/171815459-bace6395-5f00-4aba-98c8-f8bf32807ea2.png">

In this case, it definitely refers to the second point 😉 

It basically means that it accepts a very loose definition of a `LocalDate` - even if there's time information or something else, it will just truncate that into a `LocalDate`.

Reply:
Hahaha til. And I was *really* wondering how the hell an adapter can be promiscuous 😂

Aha is the use case basically to not fail to parse this exact case?
https://pl.kotl.in/wAn0op3R6
I can totally see it then. I will make the new adapter! Thanks for pointing it out! What an eagle eye spotting that 👀

Reply:
Moreso like this: https://pl.kotl.in/otOfxjDkh

Reply:
Yeah I was referring to how this breaks if there is also hour and minute etc information (which there is coming from the backend as I understand), that's what I tried to convey with my example as well :D